packages/layout/src/FlexGrid.ts

```diff
@@ -0,0 +1,139 @@
+            })
+            .each(function() {
+                const rect = this.getBoundingClientRect();
+                cachedSizes.push([
+                    rect.width,
+                    rect.height
+                ]);
+            })
+            .each(function(w, i) {
+                this.firstChild.style.display = "block";
+                w.resize({
+                    width: cachedSizes[i][0] - (2 * context.borderWidth()),
+                    height: cachedSizes[i][1] - (2 * context.borderWidth())
+                });
+            })
+            ;
+    }
+    exit(domNode, element) {
+        super.exit(domNode, element);
+    }
+    updateFlexParent(element) {
+        element
+            .style("height", "100%")
+            .style("flex-direction", this.orientation() === "horizontal" ? "row" : "column")
+            .style("flex-wrap", this.flexWrap())
+            .style("align-items", this.alignItems())
+            .style("align-content", this.alignContent())
+            .style("overflow-x", () => {
+                if (this.orientation() === "horizontal" && this.flexWrap() === "nowrap" && this.allowScroll()) {
+                    return "scroll";
+                }
+                return "hidden";
+            })
+            .style("overflow-y", () => {
+                if (this.orientation() === "vertical" && this.flexWrap() === "nowrap" && this.allowScroll()) {
+                    return "scroll";
+                }
+                return "hidden";
+            })
+            ;
+    }
+}
+FlexGrid.prototype._class += " layout_FlexGrid";
+
+export interface FlexGrid {
+    widgets(): any;
+    widgets(_: any): this;
+    orientation(): "horizontal" | "vertical";
+    orientation(_: "horizontal" | "vertical"): this;
+    flexWrap(): "nowrap" | "wrap" | "wrap-reverse";
+    flexWrap(_: "nowrap" | "wrap" | "wrap-reverse"): this;
+    itemMinHeight(): number;
+    itemMinHeight(_: number): this;
+    itemMinWidth(): number;
+    itemMinWidth(_: number): this;
+    alignItems(): "flex-start" | "center" | "flex-end" | "stretch";
+    alignItems(_: "flex-start" | "center" | "flex-end" | "stretch"): this;
+    alignContent(): "flex-start" | "center" | "flex-end" | "stretch" | "space-between" | "space-around";
+    alignContent(_: "flex-start" | "center" | "flex-end" | "stretch" | "space-between" | "space-around"): this;
+    borderColor(): string;
+    borderColor(_: string): this;
+    borderWidth(): number;
+    borderWidth(_: number): this;
+    flexGrow(): number;
+    flexGrow(_: number): this;
+    widgetsFlexGrow(): number[];
+    widgetsFlexGrow(_: number[]): this;
+    flexBasis(): string;
+    flexBasis(_: string): this;
+    widgetsFlexBasis(): string[];
+    widgetsFlexBasis(_: string[]): this;
+    allowScroll(): boolean;
+    allowScroll(_: boolean): this;
+
+}
+
+FlexGrid.prototype.publish("borderColor", "transparent", "html-color", "Color of list item borders");
```

Comment:
Is this the border on each list item - or the border on the whole widget?  If item - them maybe "itemBorderColor" would be more consistent with "itemMinHeight" etc.?